aten/src/ATen/native/LinearAlgebra.cpp

```diff
@@ -24,6 +24,20 @@
 #include <numeric>
 
 namespace at {
+namespace meta {
+TORCH_META_FUNC(addmm)(const Tensor& self, const Tensor& mat1, const Tensor& mat2, const Scalar& beta, const Scalar& alpha) {
+  TORCH_CHECK(mat1.dim() == 2, "mat1 must be a matrix, got ", mat1.dim(), "-D tensor");
+  TORCH_CHECK(mat2.dim() == 2, "mat2 must be a matrix, got ", mat2.dim(), "-D tensor");
+
+  auto names = at::namedinference::propagate_names_for_addmm(mat1, mat2, self);
+  set_output(0, IntArrayRef({mat1.sizes()[0], mat2.sizes()[1]}), {}, self.options(), names);
+  auto result = maybe_get_output(0);
```

Comment:
@ngimel I forgot... did I sign up to fix this in structured kernels itself 😂 🤣 (this is OK for now but shouldn't be necessary as structured kernel's `set_output` should do this test automatically)

Reply:
We didn't figure out robust way of deduplicating size checks in structured kernel's `set_output` and TensorIterator right away, and I haven't looked at it yet in more detail.

Reply:
ok filed an issue for this https://github.com/pytorch/pytorch/issues/57827